docs(search): expand advanced search guide (#16)
CERNDocumentServer/docs-cds-rdm: docs/search/search.md

# How to search

> This page explains how to use advanced search queries in CDS and how to narrow results with the filters in the search interface.

Reviewer: please check if there are any duplications with search page, if yes, I would replace the duplication with the link to search page to avoid maintaining both

Reviewer 2: Should we be moving all that stuff here as mentioned in this ticket?
https://github.com/CERNDocumentServer/cds-rdm/issues/93

Reviewer: what do you mean by "here?" I would say not everything from https://github.com/CERNDocumentServer/cds-rdm/issues/93 fits in the search guide.

But yes, please address https://github.com/CERNDocumentServer/cds-rdm/issues/93 (thanks for finding it!), probably needs to be distributed in appropriate documentation sections 

f.e. 
"why e-groups do not receive notifications?" , "Which resource type should I use?"
feels like FAQ

etc.

Reviewer 2: In the Search section, it mentions 
```
Move search content from https://repository.cern/help/search to https://repository.cern/docs/search/search/
```

If you only need a quick search, start with a few words in the search bar.

Example result for `quantum optics`:

> ![Simple search results](images/simple-search-results.png)

Reviewer 3: ```suggestion
```

Use the options below when you want to be more precise.

> ## Advanced queries

Reviewer 3: Move "filters" here

> Advanced queries help you search more precisely by using operators, exact phrases, field names, and ranges.

Reviewer 3: ```suggestion
Advanced queries help you search more precisely by using certain special characters or phrases, field names, and ranges.
```

### Search for one or more words

> Use this when you want to broaden or narrow a simple keyword search without moving to filters yet.

Reviewer 3: ```suggestion
```

You can search for one word or several words.

Example:

- `open science`

> This type of search looks for the words in any searchable field.

Reviewer 3: ```suggestion
This type of search looks for the words.
```

If you want to require both words, use either `+` or `AND`.

Examples:

- `+open +science`
- `open AND science`

Example result for `illuminating AND light`:

> ![Search results for a query using AND](images/and-search-results.png)

Reviewer 3: Use image captions https://squidfunk.github.io/mkdocs-material/reference/images/#image-captions

If you want to exclude a word, use either `-` or `NOT`.

Examples:

- `-open +science`
- `NOT open AND science`

Example result for `illuminating NOT light`:

![Search results for a query using NOT](images/not-search-results.png)

### Search for an exact phrase

Use this when you know the exact wording of a title, name, or expression.

Use quotation marks when you want CDS to search for the exact phrase.

Examples:

- `"open science"`

> - `"Illuminating Light"`

Reviewer 3: ```suggestion
```

Example result for `"Illuminating Light"`:

![Search results for an exact phrase query](images/exact-phrase-search-results.png)

> This is useful when you know the exact title, or when a normal keyword search returns too many unrelated results.

Reviewer 3: ```suggestion
```

### Search in a specific field

Use this when you know which part of the record you want to search, such as the title, creator, or publisher.

You can search in a specific field by writing the field name, followed by a colon, followed by your search term.

Example result for `title:illuminating`:

![Search results for a field-specific query](images/field-search-title-results.png){ width="700" }

Examples:

| Search | What it does |
| --- | --- |
| `title:open` | searches for `open` in the title |
| `title:"open science"` | searches for the exact phrase in the title |
| `creator:"ATLAS Collaboration"` | searches for a creator name |
| `publisher:CERN` | searches for the publisher |
| `description:policy` | searches in the description |
| `language:en` | searches for records in English |
| `publication_date:2024-01-01` | searches for a specific publication date |

CDS also supports shorter field names for several common identifiers and CERN-specific fields.

> Examples:

Reviewer 3: combine the two tables, remove this text

| Search | What it does |
| --- | --- |
| `doi:"10.17181/example-doi"` | searches for a DOI |
| `cds:12345` | searches for a CDS identifier |
| `cdsrn:CERN-THESIS-2021-078` | searches for a CDS report number |
| `experiment:ATLAS` | searches for an experiment |
| `department:EP` | searches for a department |

When you search for a DOI, put the DOI value in double quotes.

Example result for `doi:"10.17181/rpsnb-kjc56"`:

> ![Search results for a DOI query](images/doi-search-results.png)

Reviewer 3: ```suggestion
```

### Combine different query types

> Use this when one search idea is not specific enough on its own.

Reviewer 3: ```suggestion
```

You can combine simple terms, exact phrases, and field searches in the same query.

Examples:

- `title:"open science" AND creator:CERN`
- `title:(open science)`
- `+title:"open science" -description:policy`

Example result for `creator:user AND title:"illuminating light"`:

![Search results for a combined query](images/combined-query-results.png)

> Use parentheses when you want to keep terms together inside the same field.

Reviewer 3: Move above the image

### Search by date range

Use this when you want records from a specific period.

Use a range query when you want records within a date interval.

Examples:

- `publication_date:[2017 TO 2018]`
- `publication_date:[2017-01-01 TO 2017-12-31]`
- `publication_date:{* TO 2017-01-01}`

> - `publication_date:[2017-01-01 TO *]`

Reviewer 3: add a `[}` example

Example result for `publication_date:[2017 TO 2018]`:

![Search results for a date range query](images/date-range-search-results.png)

Rules to remember:

- `TO` must be written in capital letters.
- Square brackets `[]` mean the range is inclusive.
- Curly brackets `{}` mean the range is exclusive.
- You can use `*` for an open-ended range.

### Search for missing values

> Use this when you need to find records that include, or do not include, a certain field.

Reviewer 3: ```suggestion
```

You can search for records that have, or do not have, a value in a field.

Examples:

- `_exists_:metadata.creators`
- `NOT _exists_:metadata.additional_titles`

> You do not need the options below for most searches. Use them only when a simpler query is not enough.

Reviewer 3: ```suggestion
```

### Other supported advanced syntax

> Use this only when you already know the exact query pattern you need.

Reviewer 3: ```suggestion
```

CDS also supports several more advanced query types:

> - Regular expressions:

Reviewer 3: Make the bit before the colon bold

  `pids.doi.identifier:/10\.5281.*/`
  Use this when you want to match a pattern instead of one exact value. For example, this can help if you want to find records whose DOI starts with the same prefix.

>   Example: to search for records with a CERN DOI prefix, use `pids.doi.identifier:/10\.17181\/.*/`

Reviewer 3: move examples to sub bullets

  Example: `"large collider"~3` can help find results where the words appear near each other.
- Wildcards:
  `ope? scien*`
  Use this when you want to match small variations of a word. For example, wildcards can help if you want to search for singular and plural forms, or several words with the same beginning.
  Example: `physic*` can match words like `physics` or `physicist`.
- Boosting:
  `title:"open science"^5 description:"open science"`
  Use this when you want one part of the query to matter more than another. For example, you might want matches in the title to rank higher than matches in the description.
  Example: `title:"machine learning"^5 description:"machine learning"` gives more weight to title matches.

!!! tip
    Wildcard and regular expression searches can be powerful, but they can also be slower and harder to read. Use them only when a simpler query is not enough.

## Filters

After you run a search, you can use the filters in the left-hand panel to narrow the results without rewriting your query.

### Available filters

CDS search currently includes these filters in the user interface:

- **Publication date**: narrow results to a specific year or date range

> - **Access status**: limit results to records such as **Open**, **Embargoed**, **Restricted**, or **Metadata-only**

Reviewer 3: ```suggestion
- **Access status**: limit results to records such as **Open**, **Restricted**, or **Metadata-only**
```

- **File type**: filter by file format
- **Resource type**: filter by the kind of record, for example publication, dataset, image, or other resource categories

Example of the filters panel on a search results page:

> ![Search results page showing the filters panel](images/filters-panel-results.png)

Reviewer 3: separate into little images

### How to use filters

1. Run a search.
2. Look at the filters in the left-hand panel of the results page.
3. Apply one filter at a time to narrow the results.
4. Add more filters if you want to narrow the list further.
5. Remove a filter, or clear all filters, if the result list becomes too narrow.

Filters work together with your search query. For example, you can start with a query in the search bar and then use filters to narrow the results by publication date or resource type.

### Example of filtered results

In this example, the search results are narrowed by publication date, access status, and file type. Applying filters reduces the number of results and helps you focus on the records that match your needs more closely.

> ![Search results after applying filters](images/filtered-results-example.png){ width="650" }

Reviewer 3: remove black bar from top of image(s)

Reviewer 3: plus make the screenshot a bit wider to make it less tall
